### DISCUSSION TITLE
Random courses for online play

### DISCUSSION THREAD

**Discussion by Sonic-TV** [emoose/OutRun2006Tweaks#255] [Suggestions (Multiplayer)]
### Describe the feature you'd like to see added

You can make a random course mode since online features added.
Some examples could be
Outrun mode but every course is randomly selected (that means you don't know where you starts and you don't know what stage will be next)
Random courses 5x
Random courses 15x

**Comment by Tenome** [comment]
The course layouts are saved pre-sorted in the script BIN files, but I think I recall reading somewhere that random courses while playing was a cut feature.

**Comment by envido32** [comment]
Random courses sound like a very fun new game mode. I've been reading Howards finding and he was trying something similar (not online thou)

https://forum.arcadecontrols.com/index.php/topic,130891.msg1681190.html#msg1681190

For what I could understand the courses are all pre-loaded in memory at once but shown once you get to the exit. That means that he could change the layout on the fly by changing some memory pointers, he didn't releases a working randomizer because it was too hard to debugg with the pointers all over the place (stages, ramps, exits, goals...) but it was the basic idea behind the "Course Editor"

**Comment by Tenome** [comment]
Scripts/bin has a file called NetworkRandomCourseSP.bin, that isn't it is it? It seems to be blank when viewed in a hex editor though.

**Comment by envido32** [comment]
> Scripts/bin has a file called NetworkRandomCourseSP.bin, that isn't it is it? It seems to be blank when viewed in a hex editor though.

wait... yours is blank? Mine is very similar to `csc_data_cvt.bin` that is suppose to define the Outrun2SP tracks...

![image](https://github.com/user-attachments/assets/afa398cf-71f5-4184-8f56-fd76b87b4406)
![image](https://github.com/user-attachments/assets/b3c6dfef-eec3-49cc-b6f3-bf12a3689ba2)

**Comment by Tenome** [comment]
Nah I just didn't notice that. We have the same file.

**Comment by Zoukambada** [comment]
This is almost closing right? According to #180 ...
This currently works but maybe it's not fully tested.
To be perfect it just needs to sync/send the host "share codes" with other players.

**Comment by Zoukambada** [comment]
I found an interesting **trivia** about this game that might fit here, available at [GameFaqs section of Xbox OutRun2006: Coast 2 Coast](https://gamefaqs.gamespot.com/xbox/931342-outrun-2006-coast-2-coast):

_"Not all features the developers wanted made it into the game. According to designer Travis Ryan, "I think the hardest cuts came at the end of the project when we had to lose Create Course and Race Attack modes. In Create Course mode, you created your very own OutRun grid to race and share with friends online. You could use any stages -- mixing up normal stages, reverse stages -- from the 60 on offer to create a near-limitless variety of routes. Race Attack mode was a new arcade mode in which you tried to beat as many rivals and special rivals as possible. The team also loved the idea of being able to create your own OutRun driver, boy or girl. Maybe we will see these features in future OutRun titles..."_

This trivia is related to the other issues as well: #78 and #110

- emoose already created the beta version of "Create Course mode" exactly as the developers intended, with course codes to be shared with friends online.
- "Create your driver mode" (select alternate drivers in license screen creation) is somewhat done but not perfect
- "Race Attack mode" is the only one not featured yet in Tweaks, would it be like a survival mode from fighting games?
